File: app/src/main/java/com/sjl/bookmark/ui/adapter/CategoryAdapter.java
package com.sjl.bookmark.ui.adapter;

import androidx.annotation.Nullable;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;
import com.sjl.bookmark.R;
import com.sjl.bookmark.entity.Category;

import java.util.List;

/**
 * Android知识分类
 *
 * @author Kelly
 * @version 1.0.0
 * @filename CategoryAdapter.java
 * @time 2018/3/22 16:02
 * @copyright(C) 2018 song
 */
public class CategoryAdapter extends BaseQuickAdapter<Category, BaseViewHolder> {

    public CategoryAdapter(int layoutResId, @Nullable List<Category> data) {
        super(layoutResId, data);
    }

    @Override
    protected void convert(BaseViewHolder helper, Category item) {
        helper.setText(R.id.typeItemFirst, item.getName());
        StringBuffer sb = new StringBuffer();
        for (Category.ChildrenBean childrenBean : item.getChildren()) {
            sb.append(childrenBean.getName() + "     ");
        }
        helper.setText(R.id.typeItemSecond, sb.toString());

    }

}

File: app/src/main/java/com/sjl/bookmark/ui/adapter/CategoryAdapter.kt
package com.sjl.bookmark.ui.adapter

import com.chad.library.adapter.base.BaseQuickAdapter
import com.chad.library.adapter.base.BaseViewHolder
import com.sjl.bookmark.R
import com.sjl.bookmark.entity.Category

/**
 * Android知识分类
 *
 * @author Kelly
 * @version 1.0.0
 * @filename CategoryAdapter.java
 * @time 2018/3/22 16:02
 * @copyright(C) 2018 song
 */
class CategoryAdapter(layoutResId: Int, data: List<Category>?) : BaseQuickAdapter<Category, BaseViewHolder>(layoutResId, data) {
    override fun convert(helper: BaseViewHolder, item: Category) {
        helper.setText(R.id.typeItemFirst, item.name)
        val sb = StringBuffer()
        for (childrenBean in item.children) {
            sb.append(childrenBean.name + "     ")
        }
        helper.setText(R.id.typeItemSecond, sb.toString())
    }
}